robusta_krr/core/runner.py

```
@@ -152,6 +165,139 @@ def _process_result(self, result: Result) -> None:
+        """
+        Extracts the storage account name and container name from the SAS URL.
+        """
+        try:
+            parsed = urlparse(sas_url)
+            storage_account = parsed.hostname.split('.')[0]  # Extract the storage account name from the hostname
+            container = parsed.path.strip('/').split('/')[0]  # Extract the first part of the path as the container name
+
+            return storage_account, container
+        except Exception as e:
+            logger.error(f"Failed to extract storage info from SAS URL: {e}")
+            raise ValueError("Invalid SAS URL format. Please provide a valid Azure Blob Storage SAS URL.")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)